e2e/scenarios/kops-simple

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -x
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd ${REPO_ROOT}
+cd ..
+WORKSPACE=$(pwd)
+
+# Create bindir
+BINDIR=${WORKSPACE}/bin
+export PATH=${BINDIR}:${PATH}
+mkdir -p "${BINDIR}"
+
+# Build kOps
+pushd ${WORKSPACE}/kops
+GOBIN=${BINDIR} go install k8s.io/kops/cmd/kops
+GOBIN=${BINDIR} make test-e2e-install
+popd
+
+if [[ -z "${K8S_VERSION:-}" ]]; then
+  K8S_VERSION="$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"
+fi
+
+CLOUD_PROVIDER="gce"
+echo "CLOUD_PROVIDER=${CLOUD_PROVIDER}"
+
+if [[ -z "${CLUSTER_NAME:-}" ]]; then
+  SCRIPT_NAME=$(basename $0 .sh)
+  CLUSTER_NAME="${SCRIPT_NAME}.k8s.local"
+fi
+echo "CLUSTER_NAME=${CLUSTER_NAME}"
+
+if [[ -z "${WORKDIR:-}" ]]; then
+  WORKDIR="${WORKSPACE}/clusters/${CLUSTER_NAME}"
+fi
+mkdir -p "${WORKDIR}"
+
+if [[ -z "${GCP_PROJECT}" ]]; then
+  GCP_PROJECT=$(gcloud config get project)
+fi
+echo "GCP_PROJECT=${GCP_PROJECT}"
+
+if [[ -z "${KOPS_STATE_STORE}" ]]; then
+  KOPS_STATE_STORE="gs://kops-state-${GCP_PROJECT}"
+fi
+echo "KOPS_STATE_STORE=${KOPS_STATE_STORE}"
+export KOPS_STATE_STORE
+
+if [[ -z "${IMAGE_REPO}" ]]; then
+  IMAGE_REPO="gcr.io/${GCP_PROJECT}"
+fi
+echo "IMAGE_REPO=${IMAGE_REPO}"
+
+cd ${REPO_ROOT}
+if [[ -z "${IMAGE_TAG}" ]]; then
+  IMAGE_TAG=$(git rev-parse --short HEAD)-$(date +%Y%m%dT%H%M%S)
+fi
+echo "IMAGE_TAG=${IMAGE_REPO}"
+
+# Build and push cloud-controller-manager
+cd ${REPO_ROOT}
+export KO_DOCKER_REPO="${IMAGE_REPO}"
+ko build --tags ${IMAGE_TAG} --base-import-paths --push=true ./cmd/cloud-controller-manager/
+
```

Comment:
I guess we want to build and push external credential provider - single binary which is used by kubelet as a shortliving process that generates GCR credentials.

This probably should be something like `ko build ./cmd/auth-provider-gcp`

Reply:
Yes, good idea.  Do you know if we have any e2e tests (anywhere) that would fail if we did this "wrong"?   We do have a handful of e2e tests in the kOps repo itself, so the good news is that we don't have to get these upstream into k/k, we can easily `go test` some tests from this repo.

Reply:
I know https://testgrid.k8s.io/google-gce#disable-cloud-provider&width=20 had a pretty broad set of failures before we got the pieces running correctly...

Reply:
We are currently passing Conformance (skipping Serial) - I wonder if these tests would have to be built outside of k/k (because they are cloud provider specific)